.github/workflows/build-ui-and-server.yml

```diff
@@ -91,41 +91,43 @@ jobs:
         uses: actions/upload-artifact@v4
```

Comment:
Moving the artifact pull above, in case it fails, no need to wait for all the Kubernetes cluster and updater to be up and running